File: app/src/main/java/com/silentgames/silent_planet/model/cells/onVisible/Arrows/Red.java
package com.silentgames.silent_planet.model.cells.onVisible.Arrows;

import com.silentgames.silent_planet.R;
import com.silentgames.silent_planet.model.GameMatrixHelper;
import com.silentgames.silent_planet.utils.BitmapEditor;

/**
 * Created by Lantiets on 28.08.2017.
 */

public class Red extends Arrow {

    public Red() {
        super.setBitmap(BitmapEditor.getCellBitmap(R.drawable.arow_red_cell));
        super.setCanMove(true);
        super.setRotateAngle(BitmapEditor.RotateAngle.DEGREES0);
        super.setDistance(3);
    }

    public Red rotate(int x, int y, BitmapEditor.RotateAngle rotateAngle){
        switch (rotateAngle){
            case DEGREES0:
                setDestinationX(x);
                setDestinationY(y - getDistance());
                break;
            case DEGREES90:
                setBitmap(BitmapEditor.rotateBitmap(BitmapEditor.RotateAngle.DEGREES90,getBitmap()));
                setDestinationX(x + getDistance());
                setDestinationY(y);
                break;
            case DEGREES180:
                setBitmap(BitmapEditor.rotateBitmap(BitmapEditor.RotateAngle.DEGREES180,getBitmap()));
                setDestinationX(x);
                setDestinationY(y + getDistance());
                break;
            case DEGREES270:
                setBitmap(BitmapEditor.rotateBitmap(BitmapEditor.RotateAngle.DEGREES270,getBitmap()));
                setDestinationX(x - getDistance());
                setDestinationY(y);
                break;
        }
        setRotateAngle(rotateAngle);
        return this;
    }
}

File: app/src/main/java/com/silentgames/silent_planet/model/cells/onVisible/Arrows/Red.kt
package com.silentgames.silent_planet.model.cells.onVisible.Arrows

import com.silentgames.silent_planet.R
import com.silentgames.silent_planet.model.GameMatrixHelper
import com.silentgames.silent_planet.utils.BitmapEditor

/**
 * Created by Lantiets on 28.08.2017.
 */

class Red : Arrow() {
    init {
        super.bitmap = BitmapEditor.getCellBitmap(R.drawable.arow_red_cell)
        super.isCanMove = true
        super.rotateAngle = BitmapEditor.RotateAngle.DEGREES0
        super.distance = 3
    }

    override fun rotate(x: Int, y: Int, rotateAngle: BitmapEditor.RotateAngle): Red {
        when (rotateAngle) {
            BitmapEditor.RotateAngle.DEGREES0 -> {
                destinationX = x
                destinationY = y - distance
            }
            BitmapEditor.RotateAngle.DEGREES90 -> {
                bitmap = BitmapEditor.rotateBitmap(BitmapEditor.RotateAngle.DEGREES90, bitmap)
                destinationX = x + distance
                destinationY = y
            }
            BitmapEditor.RotateAngle.DEGREES180 -> {
                bitmap = BitmapEditor.rotateBitmap(BitmapEditor.RotateAngle.DEGREES180, bitmap)
                destinationX = x
                destinationY = y + distance
            }
            BitmapEditor.RotateAngle.DEGREES270 -> {
                bitmap = BitmapEditor.rotateBitmap(BitmapEditor.RotateAngle.DEGREES270, bitmap)
                destinationX = x - distance
                destinationY = y
            }
        }
        this.rotateAngle = rotateAngle
        return this
    }
}
